src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -353,15 +382,10 @@ public void Read(Span<PinValuePair> pinValuePairs)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="GpioController"/> class that will use the specified numbering scheme.
-        /// The controller will default to use the driver that best applies given the platform the program is executing on.
+        /// Tries to create the GPIO driver that best matches the current hardware
         /// </summary>
-        /// <param name="numberingScheme">The numbering scheme used to represent pins provided by the controller.</param>
-        public GpioController(PinNumberingScheme numberingScheme)
-            : this(numberingScheme, GetBestDriverForBoard())
-        {
-        }
-
+        /// <returns>An instance of a GpioDriver that best matches the current hardware</returns>
+        /// <exception cref="PlatformNotSupportedException">No matching driver could be found</exception>
         private static GpioDriver GetBestDriverForBoard()
```

Comment:
Having this private prevents adding new APIs, but requires copy-pasting the implementation to the Board infrastructure. That's not so big of a deal for now, except that it requires a few ugly try-catch attempts in testing which driver is the right one. 
I think we need the identification logic to become public somehow (or at least provide each driver with a IsCompatible check)

Reply:
Let's refrain from fixing this in this PR and perhaps suggest GpioDriver.Create in a separate PR/issue

Reply:
I agree to do that later. We should just not forget about it.